jenkins/Jenkinsfile-blossom.premerge

```diff
@@ -165,22 +165,28 @@ pipeline {
 
             steps {
                 script {
-                    // Retrieve PROJECT_VER from version-def.sh, e.g, '<major>.<minor>.<patch>-SNAPSHOT'
-                    PROJECT_VER = sh(returnStdout: true, script: "bash $JENKINS_ROOT/version-def.sh | cut -d ',' -f 3 | cut -d ' ' -f 3")
-                    PROJECT_VER = PROJECT_VER.split('-')[0] // Remove trailing '-SNAPSHOT'
-                    echo PROJECT_VER
-
-                    def versions = PROJECT_VER.split('\\.')
-                    major_ver = versions[0].toInteger()
-                    minor_ver = versions[1].toInteger()
-
-                    if (major_ver >= 22) {
-                        PREMERGE_CI_2_ARGUMENT = "ci_2"
-                    } else {
-                        error("Unsupported major version: $major_ver")
-                    }
+                    container('cpu') {
+                        // Retrieve PROJECT_VER from version-def.sh, e.g, '<major>.<minor>.<patch>-SNAPSHOT'
+                        PROJECT_VER = sh(returnStdout: true,
+                                        script: '''#!/bin/bash
+                                                source jenkins/version-def.sh >& /dev/null
+                                                echo $PROJECT_VER
+                                                ''').trim()
+                        PROJECT_VER = PROJECT_VER.split('-')[0] // Remove trailing '-SNAPSHOT'
+                        echo PROJECT_VER
+
+                        def versions = PROJECT_VER.split('\\.')
+                        major_ver = versions[0].toInteger()
+                        minor_ver = versions[1].toInteger()
+
+                        if (major_ver >= 22) {
+                            PREMERGE_CI_2_ARGUMENT = "ci_2"
+                        } else {
+                            error("Unsupported major version: $major_ver")
+                        }
```

Comment:
Is there any point to keeping this conditional logic anymore?  Not seeing a realistic case where major_ver < 22 is going to be true, seems we could always echo "ci_2" here.

Reply:
yeah, we can drop this legacy compatibility support logic now